### ISSUE TITLE
Spectator game still recorded (as 'None') even though setting is unchecked

### ISSUE THREAD

**Issue by culinko** [Epix37/Hearthstone-Deck-Tracker#829]
Hi, I decided to stop bothering you on reddit and finally made an account here :)

The issue is, even though I have unchecked tracking games in spectator mode, the tracker still creates the game as 'None'. It even assigns a deck in PlayerDecks.xml.

Here is the log: http://pastebin.com/raw.php?i=ejSx61y6

**Comment by Epix37** [comment]
I will have to investigate this a bit further, but they *should* usually be removed from stats again once the games gets back to the main menu. Do they not (consistently)?

**Comment by GiggliG** [comment]
I see this happen from time to time with Innkeeper games. Sometimes they get removed automatically, sometimes i have to do it manually.

**Comment by culinko** [comment]
So, I just did some testing today and here are the results:

i went to spectate ranked: 
joined existing game - not tracked
automatically joined new game - tracked

then got to spectate arena:

joined existing match - tracked
automatically joined new match - tracked twice (checked off "auto deck selection" in the middle of the match)

joined existing game - tracked
automatically joined new game - tracked
automatically joined new game and went to main menu after turn two - tracked twice (and as a Win)

So all of the games that have "tracked" were indeed still tracked and stayed there until I restarted the tracker. After the restart, only 2 games remained there and it were the first two, because they had decks assigned to them automatically. Also if you spectate the arena it can still assign you a ranked deck if you have "auto deck selection" enabled. I think I will not use the auto deck selection from now on, but it's still a mystery why those matches were tracked anyway.

Here is the complete log: http://pastebin.com/raw.php?i=hx81FfaE

**Comment by Epix37** [comment]
Interesting.

The main problem is still that those games were tracked as "none" (right?). Which means... Do you have record "other" turned on? That includes "none" iirc. Disabling that might already fix this.

...Which means... the problem is that spectator mode detection is not working (good enough). I will try to improve that.

**Comment by culinko** [comment]
Yeah, that's the main problem, they shouldn't be tracked at all. I have only Ranked/Arena/Casual checked. This is how it looked btw (the games with mode 'None' are the spectator games): 
![spectate](https://cloud.githubusercontent.com/assets/11700272/6947674/1fcc0c4a-d8a9-11e4-83b6-45001d786633.png)

**Comment by Epix37** [comment]
Hm :/. For now you will have to live with either deleting the games every now and then (sort by none, select all of mode:none, delete) or just close hdt while spectating. I will I can figure this out soon.

**Comment by Epix37** [comment]
Alright, pretty sure I got it fixed now.
